src/components/CreateModal/CreateModal.connector.js

```diff
@@ -15,9 +15,15 @@ export const mapDispatchToProps = (dispatch, props) => {
   const closeUrl = postId
     ? postUrl(postId, urlParams, querystringParams)
     : addQuerystringToPath(baseUrl(urlParams), querystringParams)
+  const closeModal = () => dispatch(push(closeUrl))
 
   return {
-    closeModal: () => dispatch(push(closeUrl))
+    closeModal,
+    confirmLeaveCreateModal: () => {
```

Comment:
this works great, but could be slightly less code and cleaner by just passing an isDirty parameter into closeModal, then you wouldn't need two functions